stdlib/ballerina-builtin/src/main/ballerina/io/Package.md

```diff
@@ -0,0 +1,137 @@
+## Package Overview
+
+This package is designed to support input and output operations via channels in a canonical way, either in a blocking,
+or non-blocking manner.
+
+![architecture](resources/package-architecture.svg)
+
+### Channels
+A channel represents an I/O source or sink of some bytes, characters, or records that are opened for reading, or
+writing, respectively.
+### Byte channels
+The most primitive channel is the `ByteChannel` which reads and writes 8-bit bytes.
+
+```ballerina
+	// Open a file in read-write mode.
+io:ByteChannel byteChannel = io:openFile("some/file.txt", io:RW);
+
+// Read 100 bytes from the file.
+var readResult = byteChannel.read(100);
+    match readResult {
+        (blob, int) content => {
+            var (bytes, _) = content;
+            return bytes; // Return the read bytes.
+        }
+    error err => {
+        return err; // An IO error occurred when reading the bytes.
+    }
+}
+
+// Write some content to the beginning of the file.
+var content = "some content".toBlob("UTF-8");
+var writeResult = byteChannel.write(content, 0);
+match writeResult {
+    int numberOfBytesWritten => {
+        return numberOfBytesWritten; // Return the count of written bytes.
+    }
+    error err => {
+        return err; // An IO error occurred when writing the bytes.
+    }
+}
+```
+### Character channels
+The `CharacterChannel` is used to read and write characters. The charset encoding is specified when creating the
+`CharacterChannel`.
+
+ ```ballerina
+// Create a `CharacterChannel` from the `ByteChannel`.
+var charChannel = new io:CharacterChannel(byteChannel, "UTF-8");
+```
+
+If a `CharacterChannel` points to a JSON or XML source, it can be read and then written, directly into a variable of
+the respective type.
+
+```ballerina
+// Reading a JSON.
+var readResult = charChannel.readJson();
+match readResult {
+    json value => {
+        return value; // Return the read JSON value.;
+    }
+    error err => {
+        return err; // An IO error occurred when reading the JSON.
+    }
+}
+
+// Writing a JSON.
+json content = {fname:"Jhon", lname:"Doe", age:40};
+var writeResult = charChannel.writeJson(content);
+match writeResult {
```

Comment:
I presume we also need to match the nill here, this is the current sample you might be able to reuse 

```
    match characterChannel.writeJson(content){
          error err =>{
            throw err;
          }
          () =>{
            io:println("Content written successfully");
          }
    }
```

Reply:
fixed in d492f4ce8c589ca785319da96c38c9df63ceee82